lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -514,6 +514,47 @@ struct ExprVisitor {
     return result;
   }
 
+  // Handle conditional operator `?:`.
+  Value visit(const slang::ast::ConditionalExpression &expr) {
+    auto type = context.convertType(*expr.type);
+
+    // Handle condition.
+    Value cond = convertToSimpleBitVector(
+        context.convertExpression(*expr.conditions.begin()->expr));
+    cond = convertToBool(cond);
+    if (!cond)
+      return {};
+    auto conditionalOp = builder.create<moore::ConditionalOp>(loc, type, cond);
+
+    // Create blocks for true region and false region.
+    conditionalOp.getTrueRegion().emplaceBlock();
+    conditionalOp.getFalseRegion().emplaceBlock();
+
+    OpBuilder::InsertionGuard g(builder);
+
+    // Handle left expression.
+    builder.setInsertionPointToStart(conditionalOp.getBody(0));
+    auto trueValue = context.convertExpression(expr.left());
+    if (!trueValue)
```

Comment:
In this case you should just `return {}`, since `convertExpression` returning a null value indicates that an error has happened that has already been reported in a diagnostic by the expression conversion code. Similar to how you further above check `if (!cond) return {};`.

Reply:
If just `return {}`, it will return an unexpected error that "error: 'moore.conditional' op expects a non-empty block".

Reply:
That is very strange! This means that even though you returned `{}` here, the Slang-to-MLIR conversion continued and at the end the MLIR verifier was called to make sure that the IR is correct. We might be missing additional checks for `if (!<expr>) return {};` elsewhere in the converter.

In theory what should happen is: when you encounter an error and the `convert*` functions start returning `{}`, this abort and return with `{}` should propagate all the way to the top, and in ImportVerilog, conversion should be aborted without the MLIR verifier running at all.

Do you have the input Verilog that causes this issue? I can try to help you find why this happens :smiley:

Reply:
Sure! Here is the input:
```
module top();
  int a,b;
  initial begin
    a = b ? 'x : b;
  end
endmodule
```

Reply:
Great catch! It looks like there was a very subtle bug in `convertStatement` where a failure from `convertExpression` would not be propagated properly. That was also the reason why we were able to check for multiple errors in `errors.sv` in parallel, which was suspicious. (Import should have aborted after first error.)

I have pushed a fix straight to the main branch: https://github.com/llvm/circt/commit/dd68e3c6f4dd1b503d3dc15a68df1d74d0b6d374

Reply:
Nice! Fixing this bug has been of great help to me. 🥰